packages/xo-web/src/xo-app/home/index.js

```diff
@@ -749,7 +749,10 @@ export default class Home extends Component {
               map(
                 tags,
                 tag =>
-                  new ComplexMatcher.RegExp(`^${escapeRegExp(tag.id)}$`, 'i')
+                  new ComplexMatcher.RegExp(
+                    `^${escapeRegExp(typeof tag === 'string' ? tag : tag.id)}$`,
+                    'i'
+                  )
```

Comment:
Can you explain this change?

Reply:
Do you mean that there will be heterogeneous types (objects and strings) in the `value` array? Then why don't we create an object on the fly when `addTag` is called so that there are only objects and the API doesn't change?